src/components/navMenu/CollapseNav.vue

```diff
@@ -0,0 +1,79 @@
+<template>
+  <div>
+    <el-menu
+      :default-active="defaultActive"
+      class="el-menu-vertical-demo"
+      @open="handleOpen"
+      @close="handleClose"
+      :collapse="isCollapse"
+    >
+      <slot></slot>
+    </el-menu>
+     <transition name="button-move">
+     <el-button
+      v-bind:icon="isCollapse ? 'el-icon-arrow-right button-arrow' : 'el-icon-arrow-right button-arrow button-arrow-rotate'"
+      v-on:click="isCollapse = !isCollapse"
+      size="mini"
+      :class="!isCollapse ? 'btn-not-collapse' : ''"
+      circle
+    ></el-button>
+     </transition>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'CollapseNav',
+  props: {
+    defaultActive: {
+      type: String,
+      default: ''
+    }
+  },
+  data () {
+    return {
+      isCollapse: true
+    }
+  },
+  methods: {
+    handleOpen (key, keyPath) {
+      console.log(key, keyPath)
+    },
+    handleClose (key, keyPath) {
+      console.log(key, keyPath)
+    }
+  }
+}
+</script>
+
+<style scoped lang="scss">
+.el-menu-vertical-demo:not(.el-menu--collapse) {
+  width: 200px;
+  min-height: 400px;
+}
+
+.el-button{
+    position:relative;
+    top:-210px;
+    left:50px;
+    transition: left 0.4s;
+}
+
+.el-button.btn-not-collapse{
+    top:-385px;
+    left:187px;
+}
+
+.button-arrow-rotate {
+  transform: rotate(180deg);
+}
+
+.button-arrow {
+  transition-duration:0.5s;
+  display: inline-block;
+}
+
+.el-menu {
+  font-family: "Lato";
+}
+</style>
```

Comment:
Try this:
```
<template>
  <div>
    <el-menu
      :default-active="defaultActive"
      class="el-menu-vertical-demo"
      @open="handleOpen"
      @close="handleClose"
      :collapse="isCollapse"
    >
      <slot></slot>
      <transition name="button-move">
        <el-button
          v-bind:icon="isCollapse ? 'el-icon-arrow-right button-arrow' : 'el-icon-arrow-left button-arrow'"
          v-on:click="isCollapse = !isCollapse"
          size="mini"
          :class="!isCollapse ? 'btn-not-collapse' : ''"
          circle
        ></el-button>
      </transition>
    </el-menu>
  </div>
</template>

<script>
export default {
  name: 'CollapseNav',
  props: {
    defaultActive: {
      type: String,
      default: ''
    }
  },
  data () {
    return {
      isCollapse: true
    }
  },
  methods: {
    handleOpen (key, keyPath) {
      console.log(key, keyPath)
    },
    handleClose (key, keyPath) {
      console.log(key, keyPath)
    }
  }
}
</script>

<style scoped lang="scss">
  .el-menu {
    font-family: "Lato";

    .el-button {
      position: absolute;
      transition: left 0.4s;
      top: 15px;
      right: -15px;
      z-index: 10;

      .button-arrow {
        transition-duration: 0.5s;
        display: inline-block;
      }

      .button-arrow-rotate {
        transform: rotate(180deg) !important;
      }
    }
  }
  .el-menu:not(.el-menu--collapse) {
    width: 200px;
    /*min-height: 400px;*/
  }
</style>

```
The height of the sidenav is something which should be dynamic.

Reply:
thank youuuu @michaelbukachi!! you are a 🌟
made the changes ✅